core/lib/zksync_core/src/api_server/web3/namespaces/zks.rs

```diff
@@ -582,13 +583,31 @@ impl ZksNamespace {
             .state
             .tx_sender
             .0
-            .l1_gas_price_source
-            .estimate_effective_gas_price();
+            .batch_fee_input_provider
+            .get_batch_fee_input()
+            .l1_gas_price();
 
         method_latency.observe();
         gas_price.into()
     }
 
+    #[tracing::instrument(skip(self))]
+    pub fn get_main_node_fee_params_impl(&self) -> MainNodeFeeParams {
```

Comment:
why do we have main_node here? 
seems external node also can expose it

Reply:
they can expose it, but it is the main node that truly sets it... Okay, I'll rename it together with the config params